common/lib/xmodule/xmodule/tests/helpers.py

```diff
@@ -34,18 +34,26 @@ class StubUserService(UserService):
     Stub UserService for testing the sequence module.
     """
 
-    def __init__(self, is_anonymous=False, **kwargs):
+    def __init__(self, is_anonymous=False, user=None, **kwargs):
         self.is_anonymous = is_anonymous
+        self.user = user
         super().__init__(**kwargs)
 
     def get_current_user(self):
         """
         Implements abstract method for getting the current user.
         """
         user = XBlockUser()
-        if self.is_anonymous:
+        if self.is_anonymous or (self.user and not self.user.is_authenticated):
             user.opt_attrs['edx-platform.username'] = 'anonymous'
             user.opt_attrs['edx-platform.is_authenticated'] = False
+        elif self.user:
+            user.opt_attrs['edx-platform.anonymous_user_id'] = 'student'
```

Comment:
Since this is a stub, we should only add logic (attributes) that are being used.

Reply:
Agreed -- [`get_test_system()` used to pass it to the runtime](https://github.com/edx/edx-platform/pull/28440/files#diff-caad0b6dd6279f98a4b30df46f5bbd1b895e5167e188658de1d9ae308c06c560L146), so instead I stored it here, but I will make this an arg to StubUserService instead, see 
87dcd76 .